tests/console/commands/test_version.py

```diff
@@ -46,6 +46,9 @@ def tester(command_tester_factory: CommandTesterFactory) -> CommandTester:
         ("1.2.3beta1", "prerelease", "1.2.3b2"),
         ("1.2.3b1", "prerelease", "1.2.3b2"),
         ("1.2.3", "prerelease", "1.2.4a0"),
+        ("1.2.3", "dev", "1.2.3.dev0"),
```

Comment:
if this is actually what it is doing then it is bugged, because the dev release is earlier than the non-dev release

Reply:
I see, I'm experimenting with automatically bumping versions through ci/cd pipelines, for which the `poetry version` command could be very useful. 

I've updated the logic and the docs/tests such that the version number always moves forward.